### ISSUE TITLE
Migration might not be handling WAL corruptions correctly

### ISSUE THREAD

**Issue by gouthamve** [prometheus/tsdb#379]
See upstream issue: https://github.com/prometheus/prometheus/issues/4603

The previous WAL would just truncate if there was a corruption. From the logs:
```
prometheus2-78df54755f-sxhzd prometheus2 level=warn ts=2018-09-13T10:30:41.128060768Z caller=head.go:415 component=tsdb msg="encountered WAL error, attempting repair" err="read records: corruption in segment 48 at 78834898: unexpected checksum ae7d1bfa, expected 512fafa7"

prometheus2-78df54755f-sxhzd prometheus2 level=error ts=2018-09-13T10:30:43.6731401Z caller=main.go:617 err="opening storage failed: read WAL: repair corrupted WAL: cannot handle error"
```

makes me think that truncation is not happening currently.

**Comment by gouthamve** [comment]
Looking at this, it's not actually in the migration step, but rather something in the new WAL: https://github.com/prometheus/tsdb/blob/master/head.go#L415

**Comment by krasi-georgiev** [comment]
waiting for the user's wal and will troubleshoot locally.

**Comment by eahydra** [comment]
I have the same problem.

**Comment by gouthamve** [comment]
Hi, this has been fixed in `2.4.2` of prometheus. Please upgrade and let us know if that fixes it?

**Comment by eahydra** [comment]
> Hi, this has been fixed in `2.4.2` of prometheus. Please upgrade and let us know if that fixes it?

Yeah, upgrade latest code , the problem is resolved.

**Comment by gouthamve** [comment]
Thanks, closing!
